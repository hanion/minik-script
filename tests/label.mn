// label.mn

print("break with label");
label outer_break_test
for i := 0; i < 3; ++i {
	for j := 0; j < 3; ++j {
		if j == 2 {
			break outer_break_test;
		}
		print("j");
	}
	print("should never print");
}


print("continue with label");
label outer_continue_test
for i := 0; i < 3; ++i {
	for j := 0; j < 3; ++j {
		if j >= 1 {
			continue outer_continue_test;
		}
		print("j");
	}
	print("should never print");
}

print("break with label 2");
label outer_break_test_2
for i := 0; i < 3; ++i {
	for j := 0; j < 3; ++j {
		if i == 1 and j == 1 {
			break outer_break_test_2;
		}
		print("i",i ,"j",j);
	}
}

print("continue with label 2");
label outer_continue_test_2
for i := 0; i < 3; ++i {
	for j := 0; j < 3; ++j {
		if i % 2 == 0 {
			continue outer_continue_test_2;
		}
		print("i",i ,"j",j);
	}
}
