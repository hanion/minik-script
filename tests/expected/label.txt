break with label
j
j
continue with label
j
j
j
break with label 2
i 0.000000 j 0.000000
i 0.000000 j 1.000000
i 0.000000 j 2.000000
i 1.000000 j 0.000000
continue with label 2
i 1.000000 j 0.000000
i 1.000000 j 1.000000
i 1.000000 j 2.000000
